core/src/main/java/tc/oc/pgm/listeners/PGMListener.java

```diff
@@ -258,31 +258,71 @@ public void handleItemPickup(final PlayerPickupItemEvent event) {
     if (nearestPlayer != event.getPlayer()) event.setCancelled(true);
   }
 
+  @EventHandler
+  public void lockFireTick(final MatchLoadEvent event) {
+    event
+        .getMatch()
+        .getWorld()
+        .setGameRuleValue(GameRule.DO_FIRE_TICK.getId(), Boolean.toString(false));
+  }
+
+  @EventHandler
+  public void unlockFireTick(final MatchStartEvent event) {
+    event
+        .getMatch()
+        .getWorld()
+        .setGameRuleValue(
+            GameRule.DO_FIRE_TICK.getId(),
+            event
+                .getMatch()
+                .needModule(GameRulesMatchModule.class)
+                .getGameRule(GameRule.DO_FIRE_TICK.getId()));
+  }
+
+  @EventHandler
+  public void lockFireTick(final MatchFinishEvent event) {
+    event
+        .getMatch()
+        .getWorld()
+        .setGameRuleValue(GameRule.DO_FIRE_TICK.getId(), Boolean.toString(false));
+  }
+
   //
   // Time Lock
   // lock time before, during (if time lock enabled), and after the match
   //
   @EventHandler
   public void lockTime(final MatchLoadEvent event) {
-    event.getMatch().getWorld().setGameRuleValue(DO_DAYLIGHT_CYCLE, Boolean.toString(false));
+    event
+        .getMatch()
+        .getWorld()
+        .setGameRuleValue(GameRule.DO_DAYLIGHT_CYCLE.getId(), Boolean.toString(false));
   }
 
   @EventHandler
   public void unlockTime(final MatchStartEvent event) {
     // if there is a timelock module and it is off, unlock time
     boolean unlockTime = !event.getMatch().getModule(WorldTimeModule.class).isTimeLocked();
```

Comment:
if you set this as the value for doDaylightCycle directly on the gamerule match module (as suggested in previous comment) the logic for all of this just simplifies down to doing basically the same as you do for fire.

Reply:
I don't know what you mean by this,